app/services/LocationService.js

```diff
@@ -383,70 +423,93 @@ export default class LocationServices {
+          1000,
+        );
       } else if (!status.authorization) {
         // we need to set delay or otherwise alert may not be shown
-        // setTimeout(
-        //   () =>
-        //     Alert.alert(
-        //       languages.t('label.require_location_information_title'),
-        //       languages.t('label.require_location_information_message'),
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => BackgroundGeolocation.showAppSettings(),
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_information_title'),
+              languages.t('label.require_location_information_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => BackgroundGeolocation.showAppSettings(),
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                    //                     Settings.setIsLogging(false);
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       }
     });
   }
 
+  static isheadlessAlreadyRunning() {
+    return BackgroundGeolocation.getConfig(config => {
+      if (!config.stopOnTerminate && config.notificationsEnabled) return true;
+    });
+  }
+
   static stop() {
     // unregister all event listeners
-    PushNotification.localNotification({
-      title: languages.t('label.location_disabled_title'),
-      message: languages.t('label.location_disabled_message'),
-    });
     BackgroundGeolocation.removeAllListeners();
     BackgroundGeolocation.stop();
 
+    BackgroundGeolocation.checkStatus(({ locationServicesEnabled }) => {
+      if (!locationServicesEnabled) {
+        PushNotification.localNotification({
+          id: LOCATION_DISABLED_NOTIFICATION,
+          title: languages.t('label.location_disabled_title'),
+          message: languages.t('label.location_disabled_message'),
+        });
+      } else {
+        PushNotification.localNotification({
+          id: LOCATION_DISABLED_NOTIFICATION,
+          title: languages.t('label.location_disabled_title'),
+          message: languages.t('label.location_stopped_message'),
+        });
+      }
+    });
     isBackgroundGeolocationConfigured = false;
     SetStoreData(PARTICIPATE, 'false').then(() => {
-      // nav.navigate('LocationTrackingScreen', {});
+      //nav.navigate('LocationTrackingScreen', {});
```

Comment:
I think we can just remove this block of stale code altogether as part of the PR.